test/engine_test.go

```diff
@@ -101,3 +103,37 @@ func TestPreCompileHook(t *testing.T) {
 		t.Error("callback not called")
 	}
 }
+
+// TODO @ivokob is this test necessary, considering that the rangechecker is already tested against the test engine?
+
+func TestRangeCheck(t *testing.T) {
```

Comment:
We have implemented only for test engine. CheckCircuit runs for both test engine and actual prover.